wpilibNewCommands/src/main/native/cpp/frc2/command/CommandScheduler.cpp

```diff
@@ -209,14 +209,18 @@ void CommandScheduler::Run() {
     m_watchdog.AddEpoch(command->GetName() + ".Execute()");
 
     if (command->IsFinished()) {
+      printf("Command %s is finished\n", command->GetName().c_str());
       m_impl->endingCommands.insert(command);
+      printf("Ending command %s\n", command->GetName().c_str());
       command->End(false);
+      printf("Command %s has ended\n", command->GetName().c_str());
       for (auto&& action : m_impl->finishActions) {
         action(*command);
       }
       m_impl->endingCommands.erase(command);
 
       m_impl->scheduledCommands.erase(command);
+      printf("Erasing command %s from scheduledCommands\n", command->GetName().c_str());
```

Comment:
Why print out messages?

Reply:
it is in the PR's message, that if you were to look at the unit tests, the C++ code doesn't function like the test expects it to do so. This is why I have added these print statements as a separate commit. They will be reverted before the merge